dev/archery/archery/integration/datagen.py

```diff
@@ -1557,82 +1557,102 @@ def _temp_path():
         generate_primitive_case([0, 0, 0], name='primitive_zerolength'),
 
         generate_primitive_large_offsets_case([17, 20])
+        .skip_category('csharp')
         .skip_category('Go')
         .skip_category('JS'),
 
         generate_null_case([10, 0])
+        .skip_category('csharp')
+        .skip_category('Go')    # TODO(ARROW-7901)
         .skip_category('Go')    # TODO(ARROW-7901)
         .skip_category('JS'),   # TODO(ARROW-7900)
 
         generate_null_trivial_case([0, 0])
+        .skip_category('csharp')
         .skip_category('Go')    # TODO(ARROW-7901)
         .skip_category('JS'),   # TODO(ARROW-7900)
 
         generate_decimal128_case()
         .skip_category('Rust'),
 
         generate_decimal256_case()
+        .skip_category('csharp')
```

Comment:
According to https://github.com/apache/arrow/blob/master/docs/source/status.rst#data-types, Decimal256 should be supported?

Reply:
It is supported. However, I needed to cut things off somewhere in order to get the initial implementation out. If it is easy to implement now that I have everything working, I'll implement it.

Reply:
It was pretty simple to implement this now, so I did.